src/codeu/chat/common/User.java

```diff
@@ -104,4 +122,99 @@ private static int hash(User user) {
+			  // if the last message in the conversation was sent before lastUpdateTime 
+			  // that means it was covered in the last status update and we can move on 
+			  continue;
+		  }
+		  
+		  if (convo.conversation.creation.inMs() > this.lastUpdateTime && 
+				  this.userInterests.contains(convo.conversation.owner.id())){ 
+			  // if the convo was created by a user of interest, it will be added 
+			  // to that user's value's CONVOS_CREATED_ARRAY
+			  userUpdates.get(convo.conversation.owner.id()).get(CONVOS_CREATED_ARRAY).add(convo.conversation.title); 
+		  }
+		  
+		  
+		  for (MessageContext currentmessage = convo.firstMessage(); 
+				  currentmessage != null; currentmessage = currentmessage.next()){ 
+			  // goes through all the messages in the conversation starting from the first one 
+			  if (currentmessage.message.getCreationTime() > this.lastUpdateTime){
+				  // if the currentmessage was created after the last update time 
+				  // we need to look at it to see if it was contributed by a user 
+				  // of interest. 
+				  // If so, we add it to the user's CONVOS_CONTRIBUTED_TO_ARRAY, 
+				  // otherwise we continue iterating through the messages in the convo 
+				  if (this.userInterests.contains(currentmessage.message.author) && 
+						  !userUpdates.get(convo.conversation.owner.id()).get(CONVOS_CONTRIBUTED_TO_ARRAY).contains(convo.conversation.title)){
+					  // Also checks to make sure that conversation hasn't already been added to CONVOS_CONTRIBUTED_TO_ARRAY
+					  userUpdates.get(convo.conversation.owner.id()).get(CONVOS_CONTRIBUTED_TO_ARRAY).add(convo.conversation.title); 
+				  }
+			  }
+		   }	  
+	  	}
+	  // updates the lastUpdateTime for the user 
+	  this.lastUpdateTime = updateTime.inMs(); 
+	  return userUpdates; 
+  	}
+  
+  
+  // CONVOstatusUpdate 
+  //
+  // returns the conversation updates in a dictionary 
+  // where the name of the convo is the key 
+  // and the number of messages sent since the last update is 
+  // the value 
+  // 
+  public HashMap<String, Integer> CONVOstatusUpdate(Iterable<ConversationContext> conversations){ 
+	  HashMap<String, Integer> convoUpdates = new HashMap<String, Integer>(); 
+	  int totalNumMessages, numNewMessages; 
+	  
+	  for (ConversationContext convo : conversations) {
+		  // iterates through all the convos
+		  if (this.conversationInterests.keySet().contains(convo.conversation.id)) { 
+			  // if the convo is one of interest, 
+			  // first add the conversation and the number of messages 
+			  // since last update to conversationInterests
+			  totalNumMessages = convo.getMessageCount(); 
+			  numNewMessages = totalNumMessages - this.conversationInterests.get(convo.conversation.id); 
+			  convoUpdates.put(convo.conversation.title, numNewMessages); 
+			  
+			  // then update the value for the number of messages at lastUpdate in conversationInterests
+			  this.conversationInterests.put(convo.conversation.id, totalNumMessages); 
```

Comment:
It overwrites the previous value. And there aren't any functions in model atm; could you clarify why we would want to add those?